### PR TITLE
Fix Zone Positions when using !zone

### PR DESCRIPTION
**Pull request by ghost**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

I went around these zones to fix the position of them, that put near city or zoneline. Few did not have a zone position so had to add. Stellar Fulcrum would put you in the BCNM, fixed that so you are outside the Battlefield. I know Chocobo Circuit not used at this time, but did update position for it. This was only for when use command !zone. I did get position with using !where

### CODE REVIEW

File: scripts/zones/Chocobo_Circuit/Zone.lua [hunk 1]
```diff
@@ -12,6 +12,9 @@ end
 
 zone_object.onZoneIn = function(player, prevZone)
     local cs = -1
+	if (player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0) then
```
Comment (Xaver-DaRed) [inline scripts/zones/Chocobo_Circuit/Zone.lua]: Please align this properly. 4 spaces, no tabs.
Also, please remove parenthesis ( ) containing the "if". Those are unneeded

File: scripts/zones/Stellar_Fulcrum/Zone.lua [hunk 2]
```diff
@@ -22,6 +22,10 @@ end
 
 zone_object.onZoneIn = function(player, prevZone)
     local cs = -1
+	if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
```
Comment (Xaver-DaRed) [inline scripts/zones/Stellar_Fulcrum/Zone.lua]: Align this properly, pls.

File: scripts/zones/Walk_of_Echoes/Zone.lua [hunk 3]
```diff
@@ -12,6 +12,9 @@ end
 
 zone_object.onZoneIn = function(player, prevZone)
     local cs = -1
+	 if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
```
Comment (Xaver-DaRed) [inline scripts/zones/Walk_of_Echoes/Zone.lua]: Align this properly, pls. No tabs. 4 spaces.

**Review by Xaver-DaRed — CHANGES_REQUESTED** [review]

File: scripts/zones/Stellar_Fulcrum/Zone.lua [hunk 4]
```diff
@@ -22,6 +22,10 @@ end
...
+        player:setPos(-519, 1, -33, 193)
+    end
+
```
Comment (Xaver-DaRed) [inline scripts/zones/Stellar_Fulcrum/Zone.lua]: There seems to be an extra whitespace here, pls remove it

**Comment by ghost** [comment]
Fix what was requested, If any others need fix I can update them

File: scripts/zones/Stellar_Fulcrum/Zone.lua [hunk 5]
```diff
@@ -55,4 +58,4 @@ zone_object.onEventFinish = function(player, csid, option)
 
 end
 
-return zone_object
+return zone_object
```
Comment (Xaver-DaRed) [inline scripts/zones/Stellar_Fulcrum/Zone.lua]: Dont remove the last line. Put it back, pls.

File: scripts/zones/Stellar_Fulcrum/Zone.lua [hunk 6]
```diff
@@ -22,11 +22,10 @@ end
 
 zone_object.onZoneIn = function(player, prevZone)
     local cs = -1
-	if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
+    if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
         player:setPos(-519, 1, -33, 193)
     end
-	
-    return cs
+	return cs
```
Comment (Xaver-DaRed) [inline scripts/zones/Stellar_Fulcrum/Zone.lua]: You moved the return, align it back, pls.

File: scripts/zones/Chocobo_Circuit/Zone.lua [hunk 7]
```diff
@@ -12,7 +12,7 @@ end
 
 zone_object.onZoneIn = function(player, prevZone)
     local cs = -1
-	if (player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0) then
+	if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
```
Comment (Xaver-DaRed) [inline scripts/zones/Chocobo_Circuit/Zone.lua]: You missed aligning the line.

**Review by Xaver-DaRed — CHANGES_REQUESTED** [review]

**Comment by ghost** [comment]
Updated them

File: scripts/zones/Stellar_Fulcrum/Zone.lua [hunk 8]
```diff
@@ -22,7 +22,10 @@ end
 
 zone_object.onZoneIn = function(player, prevZone)
     local cs = -1
-    return cs
+    if player:getXPos() == 0 and player:getYPos() == 0 and player:getZPos() == 0 then
+        player:setPos(-519, 1, -33, 193)
+    end
+   return cs
```
Comment (Xaver-DaRed) [inline scripts/zones/Stellar_Fulcrum/Zone.lua]: 4 spaces, not 3. Almost there.

**Review by Xaver-DaRed — CHANGES_REQUESTED** [review]

File: scripts/zones/Stellar_Fulcrum/Zone.lua (hunk 8 above)
Comment (ghost) [reply scripts/zones/Stellar_Fulcrum/Zone.lua]: fixed

**Review by Xaver-DaRed — APPROVED** [review]
Changes look acceptable. I recommend an squash on merge.

**Comment by ghost** [comment]
Thanks. It is a learning curve. But now I know what to look for.

I will update more zones over time.

**Comment by TeoTwawki** [comment]
Some of those default positions were chosen because that is where certain cs events will deposit the players. If changed, we need to be sure that the event then has a set pos line to player the character where these defaults _used to be_

There unfortunately were not chosen for convenience when using GM commands.

I don't even know WHICH ones might actually be safe to change..

**Comment by ghost** [comment]
I didn't think zone in would affect a cs. cause a lot them zones a cs should not place you, far as I know of.

Example 

!zone La Theine Plateau puts you right at front of Ordelle's Caves 

!zone Fei'Yin puts you right at front of Cloister of Frost 

!zone Attohwa Chasm puts you right at front of Boneyard Gully

!zone East Ronfaure puts you right at front of King Ranperre's Tomb

!zone Jugner Forest  puts you right at front of Carpenters' Landing

A lot other zones put you right in front BCNM. I don't remember cs doing that unless something changed, of where current positions are. But I will double check.

**Comment by TeoTwawki** [comment]
> I didn't think zone in would affect a cs. cause a lot them zones a cs should not place you, far as I know of.
> 
> Example
> 
> !zone La Theine Plateau puts you right at front of Ordelle's Caves
> 
> !zone Fei'Yin puts you right at front of Cloister of Frost
> 
> !zone Attohwa Chasm puts you right at front of Boneyard Gully
> 
> !zone East Ronfaure puts you right at front of King Ranperre's Tomb
> 
> !zone Jugner Forest puts you right at front of Carpenters' Landing
> 
> A lot other zones put you right in front BCNM. I don't remember cs doing that unless something changed, of where current positions are. But I will double check.

A lot isn't "all" but I KNOW some just drop you at 0,0,0 and need an audit of which do that. I feel the path forward is _never let a cs do that ever._  If a cs should drop a player at a specific point, we should setPos in its eventFinish script portion so that we don;t rely on default positioning. I don't like telling anybody not to move the default pos somewhere saner anymore than anybody likes hearing it., but we do have to make sure we don't drop you someplace you don't belong. :cry: 

edit for my edit for my edit for my...I can't talk right today, sorry I'm having a rough couple days so I hope I am communicating what I mean to and not sounding rude along with my poor grammer and typos.

**Comment by ghost** [comment]
I understand. I was just trying to update some of the zones on zone in. But I am double checking Missions and Quest to see. As mention a lot the zones I fixed should not drop you in that location that was previous. But I am looking so I can verify it.

**Comment by TeoTwawki** [comment]
I appreciate you taking the time to double check them

**Comment by ghost** [comment]
I have did some looking and I could be over looking but I am not see anything that will put in position of locations that I updated. I did make a New Character and did some mission, had a friend test also and nothing was breaking that I could see. If these mission or quest suppose put in that spot I can not find it. Still looking though

**Comment by TeoTwawki** [comment]
> I have did some looking and I could be over looking but I am not see anything that will put in position of locations that I updated. I did make a New Character and did some mission, had a friend test also and nothing was breaking that I could see. If these mission or quest suppose put in that spot I can not find it.

Can I trust that you will be around if we were to discover them later?

Honestly I don't think you can find the problem scenes by just player through a few missions and hoping to run into one that has the kind of situation we're discussing. It would likely require a real audit of a TON of the scripts.

**Comment by ghost** [comment]
I plan to remain with the project. If issue does occur I will be here to help fix. I even have backup of zones before any change in case something broke. I know where you are coming from. I am trying to locate but like mention I could be over looking. But I do know some testing does take time.

**Comment by ghost** [comment]
Going temporary close request until more work done. So do other PR for gear updates that's not in DB.

**Comment by TeoTwawki** [comment]
> Going temporary close request until more work done. So do other PR for gear updates that's not in DB.

If you use different branches there is no need to close and you can have multiple pull requests with different content of changes open at the same time. its recommended to leave your `base` branch untouched from the time you create your fork to make the creation of new branches simpler (as new branch creation can begin as a copy of an existing branch, usually `base` or in other projects `main` or `master`)

**Comment by TeoTwawki** [comment]
It occurs to me this may actually be a good candidate for throwing to the everything branch anyway, so I am going to prepare that.

**Comment by ghost** [comment]
Ah ok I didn't know that. Still new to all this. I will keep that in mind

**Comment by TeoTwawki** [comment]
make sure to reset your current base branch before starting your next pr, or it will still contain the work you did here and duplicate the changes. suggest reading up on branching and branch switching to help your workflow so you can do easily multiple in the future.

Meanwhile I have merged this into a temporary branch, so that I can still merge into our `base` later while putting it into `everything` now. This is sort of a "bleeding edge" branch where people can try things early.

**Comment by ghost** [comment]
Ok awesome. Thanks for that.

**Comment by ghost** [comment]
Teo I have been looking for good guide how reset base branch and about branches so I can make sure I do this right. Most I came across provided different information.

**Comment by TeoTwawki** [comment]
> Teo I have been looking for good guide how reset base branch and about branches so I can make sure I do this right. Most I came across provided different information.

what git client do you use? I'm in windows land so I most often either command line or use tortoisegit's context menu.

**Comment by ghost** [comment]
I do have GitHub Desktop. I do know of Windows Powershell. And Git when building the server.

**Comment by TeoTwawki** [comment]
> I do have GitHub Desktop. I do know of Windows Powershell. And Git when building the server.

github desktop makes this a pain, and I'm unable to guide you there. As far as command line, you'd want to use the [reset](https://www.w3schools.com/git/git_reset.asp) command pointing to a specific commit hash. There are 3 types of reset: soft, hard, and mixed. Hard reset erases all file changes and sets you index so you are identical to teh commit hash you gave it. Mixed will change the index, but live your files alone. So you would then see changed files waiting to be committed to your local copy. And I honestly can't think of a scenario I'd use the 3rd type.

If this were tortoise it'd be _right click -> show log -> select commit -> reset to this._

**Comment by ghost** [comment]
I would do like as command to do reset before I made any change?

git reset c9642dd938

**Comment by TeoTwawki** [comment]
yup

**Comment by ghost** [comment]
Ok thanks for that info

**Comment by ghost** [comment]
Teo thanks that helped alot so can be able to add future changes. I am learning alot with it all.

**Comment by TeoTwawki** [comment]
You're welcome.

**Pull request merged**